tests/unit/test_snap_management.py

```diff
@@ -0,0 +1,68 @@
+from unittest.mock import patch
+
+import pytest
+import snap_management
+from snap_management import _install_snap, install_ga_snap
+
+snap_spec = {
+    "strict": {
+        "amd64": {
+            "name": "grafana-agent",
+            "revision": "16",  # 0.35.0
+        },
+        "arm64": {
+            "name": "grafana-agent",
+            "revision": "23",  # 0.39.2
+        },
+    },
+    "classic": {
+        "some-arch": {
+            "name": "some-charm",
+            "revision": "123",
+        },
+    },
+}
+
+
+class TestSnapInstall:
```

Comment:
why the class?

Reply:
Completely organizational.  There used to be more here though, so I've just removed it.